ergo-lib/src/chain/address.rs

```diff
@@ -265,24 +310,44 @@ impl AddressEncoder {
     /// parse address from Base58 encoded string
     pub fn parse_address_from_str(&self, str: &str) -> Result<Address, AddressEncoderError> {
         let bytes = bs58::decode(str).into_vec()?;
-        if bytes.len() < AddressEncoder::MIN_ADDRESS_LENGTH {
+        if bytes.len() < 1 {
```

Comment:
What is the reason for this change?

Reply:
because the check for `AddressEncoder::MIN_ADDRESS_LENGTH` was moved to `unchecked_parse_address_from_bytes`. I still need a check of `>= 1` in this function though because of the line
```
let network_prefix = (bytes[0] & 0xF0).try_into()?;
```

I could check for `AddressEncoder::MIN_ADDRESS_LENGTH` in both places, but since this function only needs a header byte, I made the requirement be just `>= 1`

Reply:
Got it. I don't want to nitpick, but I find that the new check has some drawbacks comparing to using the constant:
-  "magic" number is harder to reason about;
-  with a 1-byte array the function will fail later anyway.